### PR TITLE
add .gitignore

### PR DESCRIPTION
**Pull request by Kostya-Vystorobets**

### CODE REVIEW

**Comment by didevgen** [comment]
Please fill in the description and specify the correct PR name
Cause 12 files are not about .gitignore :)

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/node_modules
+/package-lock.json
+/package.json
```
Comment (didevgen) [inline .gitignore]: package json and lock files should not be ignored.
Please read what is the purpose of lock files

File: app.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    .then(() => console.log('MongoDB connected'))
+    .catch((error) => console.log(error));
+
+const app = express();
+
+app.use(morgan('dev'));
+app.use(cors());
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
```
Comment (didevgen) [inline app.js:24]: what is the purpose of body parser (you need to know)

File: app.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const bodyParser = require('body-parser');
+const cors = require('cors');
+const morgan = require('morgan');
+const mongoose = require('mongoose');
+const userRoutes = require('./routes/user');
+const departmentRoutes = require('./routes/department');
+const employeeRoutes = require('./routes/employee');
+const router = require('./routes/user');
+
+mongoose.connect('mongodb+srv://Admin:X%40QekgrxJ7cBkau@cluster0.xw60b.mongodb.net/Corporation?retryWrites=true&w=majority')
```
Comment (didevgen) [inline app.js]: use .env (reach to Vlad to get consulted)

File: controllers/department.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+module.exports.getById = function (request, response) {
+
+}
+module.exports.deleteById = function (request, response) {
+
+}
+module.exports.getAll = function (request, response) {
+
+}
```
Comment (didevgen) [inline controllers/department.js]: it is simpler to do the following

```js

const getAll = () => {....};
.....


module.exports = {
deleteById,
getAll
}
```

File: controllers/department.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    const department = new Department({
+        name: request.body.name,
+        description: request.body.description,
+        employee: []
+    })
+    try {
+        await department.save()
+        response.status(201).json(department)
+    } catch (error) {
+        response.status(500).json({ message: error })
```
Comment (didevgen) [inline controllers/department.js]: you should have a app-wide error handler instead of the local handling

File: controllers/user.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        if (request.body.password === user.password) {
+            const token = jwt.sign({
+                email: user.email,
+                userId: user._id
+            }, 'dev-jwt', {expiresIn: '24h'})
+            response.status(200).json({
+                token: token
+            })     
+        } else {
+            response.status(401).json({message: "Password is not correct"})
```
Comment (didevgen) [inline controllers/user.js]: even if user exists you should respond with "not found"

File: models/Department.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        require: true
+    },
+    updated_at: {
+        type: String,
+        require: true
+    },
+    employees: {
+        type: Array,
+        require: true
+    }
```
Comment (didevgen) [inline models/Department.js]: you should define the relations
https://medium.com/@brandon.lau86/one-to-many-relationships-with-mongodb-and-mongoose-in-node-express-d5c9d23d93c2

File: controllers/user.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+const User = require("../models/User");
+const jwt = require('jsonwebtoken');
+
+
+module.exports.login = async function (request, response) {
+    const user = await User.findOne({email: request.body.email})
+    if (user) {
+        if (request.body.password === user.password) {
```
Comment (didevgen) [inline controllers/user.js]: are you going to [store](https://stackoverflow.com/questions/1054022/best-way-to-store-password-in-database) passwords as is ? :)

File: app.js [hunk 9]
```diff
@@ -19,8 +19,8 @@ const app = express();
 app.use(morgan('dev'));
 app.use(cors());
 
-app.use(bodyParser.urlencoded({ extended: true }));
-app.use(bodyParser.json());
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
```
Comment (didevgen) [inline app.js]: what was wrong with the bodyparser?

File: controllers/department.js [hunk 10]
```diff
@@ -22,9 +20,17 @@ module.exports.create = async function (request, response) {
         await department.save()
         response.status(201).json(department)
     } catch (error) {
-        response.status(500).json({ message: error })
+        errorHandler(response, error)
```
Comment (didevgen) [inline controllers/department.js]: you can handle errors on middleware level

File: services/departmen.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
+const Department = require('../models/Department');
```
Comment (didevgen) [inline services/departmen.js]: fix typo in this filename (departmen)

File: services/departmen.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+const getById = async (id) => {
+        const department = await Department.findById(id)
+        return department;
+}
+const deleteById = async (id) => {
+    const department = await Department.remove(id)
+    return department;
+
```
Comment (didevgen) [inline services/departmen.js]: this time you can return in inline way
return Department.remove(id)

File: services/departmen.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    return department;
+        
+}
+const create = async (body) => {
+    const department = new Department({
+        name: body.name,
+        description: body.description,
+        employee: []
+    })
+    await department.save()
```
Comment (didevgen) [inline services/departmen.js]: setup prettier for the project

File: services/departmen.js (hunk 11 above)
Comment (Kostya-Vystorobets) [reply services/departmen.js]: Done

File: services/departmen.js (hunk 13 above)
Comment (Kostya-Vystorobets) [reply services/departmen.js]: Done

File: services/departmen.js (hunk 12 above)
Comment (Kostya-Vystorobets) [reply services/departmen.js]: Done

File: app.js (hunk 3 above)
Comment (Kostya-Vystorobets) [reply app.js]: Done

File: .gitignore (hunk 1 above)
Comment (Kostya-Vystorobets) [reply .gitignore]: Thanks, corrected.

File: controllers/department.js (hunk 4 above)
Comment (Kostya-Vystorobets) [reply controllers/department.js]: Done, Thanks

**Review by didevgen — APPROVED** [review]

**Pull request merged**